solr/core/src/java/org/apache/solr/highlight/UnifiedSolrHighlighter.java

```diff
@@ -431,14 +435,33 @@ protected Locale parseLocale(String language, String country, String variant) {
 
     @Override
     protected Predicate<String> getFieldMatcher(String field) {
-      // TODO define hl.queryFieldPattern as a more advanced alternative to hl.requireFieldMatch.
 
       // note that the UH at Lucene level default to effectively "true"
       if (params.getFieldBool(field, HighlightParams.FIELD_MATCH, false)) {
         return field::equals; // requireFieldMatch
-      } else {
-        return NOT_REQUIRED_FIELD_MATCH_PREDICATE;
       }
+
+      String[] queryFieldPattern =
+          params.getFieldParams(field, HighlightParams.QUERY_FIELD_PATTERN);
+      if (queryFieldPattern != null && queryFieldPattern.length != 0) {
+
+        Collection<String> indexedFields = solrIndexSearcher.getDocFetcher().getIndexedFieldNames();
+
+        Predicate<String> predicate = null;
+        for (String f : expandWildcardsInHighlightFields(indexedFields, queryFieldPattern)) {
```

Comment:
You created a expandWildcardsInHighlightFields method whose first argument has a name storedHighlightFieldNames.  But at this line you pass indexed fields.  This is kind of suspicious.

Reply:
Fair point. https://github.com/apache/solr/pull/767/commits/d48a4a623ef5f50f13315314690bad9652b154be on #767 to drop the "stored" and "highlight" from the argument name (and to add "supplier") plus also to omit "highlight" from the method name since actually there's nothing highlighting specific about the expansion and the class name already indicates highlighting scope.